src/main/java/eu/dissco/core/datacitepublisher/kafka/KafkaConsumerService.java

```diff
@@ -0,0 +1,54 @@
+package eu.dissco.core.datacitepublisher.kafka;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import eu.dissco.core.datacitepublisher.domain.DigitalSpecimenEvent;
+import eu.dissco.core.datacitepublisher.domain.MediaObjectEvent;
+import eu.dissco.core.datacitepublisher.service.DataCitePublisherService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class KafkaConsumerService {
+
+  @Qualifier("objectMapper")
+  private final ObjectMapper mapper;
+  private final DataCitePublisherService service;
+  private final KafkaPublisherService kafkaPublisherService;
+
+  @KafkaListener(topics = "${kafka.consumer.topic.specimen}")
+  public  void getSpecimenMessages(@Payload String message) {
```

Comment:
Wondering a bit if we need two topics or if we can do one and then check on `ods:type` or something. Also wondering if we need to parse them to full java objects or if we can parse them to s JsonNode and then pick the relevant fields. On the other hand I don't think this approach causes any issues, so let's stick with it for now.

Reply:
Yeah I like this approach because we can use shared schemas but I see where you're coming from. let's see how this performs for now.